### ISSUE TITLE
[Rust] Server starting with only first word of "server.hostname" parameter

### ISSUE THREAD

**Issue by MrGraversen** [GameServerManagers/LinuxGSM#1675]
I've configured a server to launch as `My Rust Server`, but it is simply found in-game as `My`.

Here's the Hastebin: https://hastebin.com/cagufureli

From the Hastebin:

```
Command-line Parameters
=================================
./RustDedicated -batchmode +server.ip * +server.port 28015 +server.tickrate 60 +server.hostname "My Rust Server" +server.identity "rustserver"  +server.maxplayers 50 +server.worldsize 5000 +server.saveinterval 300 +rcon.web 1 +rcon.ip * +rcon.port 28016 +rcon.password "********" -logfile "/home/*/rust/log/server/rustserver-game-2017-10-29-01:19:37.log"
```

It seems like a weird issue; I used to host a Rust server from a canned gameservers hosting provider, and the `server.hostname` parameter worked fine.

**Comment by MrGraversen** [comment]
I strung together some information from other issues, and was able to come up with the following solution:

In my `lgsm/config-lgsm/rustserver/rustserverX.cfg` I inserted [line 119 from rustserver on the master branch](https://github.com/GameServerManagers/LinuxGSM/blob/9bf74b80c266c92110029a1aa6333acec1580299/Rust/rustserver#L119), and then the server was able to start without `debug` mode (something I also wasn't able to do), and the server name now appears to be set correctly.

**Comment by dgibbs64** [comment]
@UltimateByte Please can you check that the fix_rust.sh is working as intended. Thanks

**Comment by thorpj** [comment]
@MrGraversen 
I made the change you recommended (but in common.cfg)

From ./rustserver debug 
``Continue? [Y/n] Y
[ INFO ] Debug rustserver: Stopping any running servers
[ INFO ] Stopping rustserver: [Reload.gg]_Rust_Fort_Wars_Server is already stopped
[  OK  ] Debug rustserver: Starting debug
/home/rustserver/lgsm/functions/command_debug.sh: line 107: LD_LIBRARY_PATH=:/home/rustserver/serverfiles/RustDedicated_Data/Plugins/x86_64: No such file or directory
[  OK  ] Debug rustserver: Stopping debug
``

./rustserver start does not work either

**Comment by UltimateByte** [comment]
@thorpj Post details please as required here to try diagnosing anything.
https://github.com/GameServerManagers/LinuxGSM/wiki/Support

**Comment by dgibbs64** [comment]
@UltimateByte new support link :) https://linuxgsm.com/support/#guide

**Comment by UltimateByte** [comment]
Awesome! @dgibbs64

**Comment by thorpj** [comment]
Apologies @UltimateByte 
./rustserver debug works, ./rustserver start gives error below
Game: Rust
Details: https://hastebin.com/ilowizodes.sql
Error (found in a console.log file):
``
dlopen failed trying to load:
steamclient.so
with error:
steamclient.so: cannot open shared object file: No such file or directory
[S_API FAIL] SteamAPI_Init(): Sys_LoadModule failed to load: steamclient.so
[S_API FAIL] SteamAPI_Init() failed
InitServer: GameServer_Init returned false
eac_server.so [x64] :: OnUnload()
``
steamclient.so exists in $LD_LIBRARY_PATH which is $HOME/serverfiles/RustDedicatedData/Plugins/x86_64/steamclient.so (host is 64bit)

lsb_release -a: Ubuntu 16.04.4
uname -a: `Linux rust-01.gameserver 4.4.0-116-generic #140-Ubuntu SMP Mon Feb 12 21:23:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
ldd --version: `(Ubuntu GLIBC 2.23-0ubuntu10) 2.23`
tmux -V: 2.1
from a relevant server log file:
``
Couldn't initialize Steam Server (172.20.10.230)

...
...
...

Server startup complete
EasyAntiCheat Scout Shutting Down

(Filename: /home/builduser/buildslave/unity/build/artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)

EasyAntiCheat Server Shutting Down

(Filename: /home/builduser/buildslave/unity/build/artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)

[Raknet] Server Shutting Down (Shutting Down)
Unloaded plugin Rust:IO for Oxide v2.15.0 by playrust.io / dcode
``

**Comment by thorpj** [comment]
It appears that the fix was to add the debug flag

parms="$parms -debug" in common.cfg

Apparently there's an issue that Facepunch need to resolve

**Comment by UltimateByte** [comment]
Your start parameters are messed up. Whatever you did, this is wrong. The fix you applied is already applied properly by LinuxGSM. Adding "LD_LIBRARY_PATH=/home/rustserver/serverfiles/RustDedicated_Data/Plugins/x86_64" to start parameters is not correct.

Here is what a clean server looks like: https://hastebin.com/uyovunuxut

Please, review your config, make a backup of it and start clean from _common.cfg, and stop messing with start parameters if you don't know what you're doing.

**Comment by UltimateByte** [comment]
Also, bind to a real IP and update your LinuxGSM.

**Comment by thorpj** [comment]
@UltimateByte I have tested:
* **clean config**
* with the fix you recommended
* with edits to the fix so that it definitely links to the correct path

And the finally found that I needed to add the `-debug` parameter. Yes I can now remove that line, but clearly it's not a problem, as my server works with or without that line.

I'm not binding to a specific ip for $reasons. I have updated LGSM since the last commit. A few days ago.
